### PR TITLE
Notepad copy

### PR DESCRIPTION
**Pull request by timbl**
As a notepad user I want to be able to get/export an HTML copy to use elsewhere.

### CODE REVIEW

File: src/authn/authn.ts [hunk 1]
```diff
@@ -315,9 +315,9 @@ async function loadIndex (
 ): Promise<AuthenticationContext> {
   // Loading preferences is more than loading profile
   try {
-    ;(await isPublic)
+    context = await (isPublic
```
Comment (michielbdejong) [inline src/authn/authn.ts]: Why overwrite `context` which was passed in from https://github.com/solid/solid-ui/blob/master/src/authn/authn.ts#L312?
And also from https://github.com/solid/solid-ui/blob/master/src/authn/authn.ts#L314 it seems that `isPublic` is a boolean?

I'll dig into this function later today, adding unit tests, code documentation, and a usage example, so we can all understand better what the desired behaviour is here. Without that, this is very hard for me to review. Will review the rest of this PR independently first.

File: src/chat/infinite.js [hunk 2]
```diff
@@ -314,7 +314,7 @@ export function infiniteMessageArea (dom, kb, chatChannel, options) {
         UI.media.cameraButton(dom, kb, getImageDoc, tookPicture)
       )
 
-      UI.pad.recordParticipation(chatChannel, chatChannel.doc()) // participation =
+      UI.participation.recordParticipation(chatChannel, chatChannel.doc()) // participation =
```
Comment (michielbdejong) [inline src/chat/infinite.js]: I'll review this change separately, it seem unrelated to both this PR and this commit.

File: src/participation.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  } catch (e) {
+    container.appendChild(
+      UI.widgets.errorMessageBlock(
+        dom,
+        'Error recording your partipation: ' + e
+      )
+    ) // Clean up?
+  }
+  return table
+}
```
Comment (michielbdejong) [inline src/participation.js]: This whole commit seems to be about moving participation out of src/pad.js and into its own src/participation.js

In itself that looks like a good thing! But unrelated to notepad copy, so I'll move this into a separate PR. Also, in the new team we established a rule-of-thumb that we move any code we touch to typescript, and i would also like to add docs, unit tests, and examples in the process.

File: src/pad-utils.js [hunk 4]
```diff
@@ -32,36 +32,44 @@ export function xmlEncode (str) {
 export function notePadToHTML (pad, kb) {
   const chunks = getChunks(pad, kb)
   var html = '<html>\n  <head>\n'
-  const title = kb.anyValue(pad, ns.DCT('title'))
+  const title = kb.anyValue(pad, ns.dct('title'))
```
Comment (michielbdejong) [inline src/pad-utils.js]: this change seems unrelated?

File: src/authn/authn.ts (hunk 1 above)
Comment (michielbdejong) [reply src/authn/authn.ts]: Moving this to https://github.com/solid/solid-ui/issues/209

File: src/chat/infinite.js (hunk 2 above)
Comment (michielbdejong) [reply src/chat/infinite.js]: Continued in https://github.com/solid/solid-ui/issues/210

File: src/index.ts [hunk 5]
```diff
@@ -59,6 +59,8 @@ import { infiniteMessageArea } from './chat/infinite'
 // @ts-ignore
 import pad from './pad'
 // @ts-ignore
+import participation from './participation'
+// @ts-ignore
```
Comment (michielbdejong) [inline src/index.ts]: Moved to https://github.com/solid/solid-ui/issues/210

File: src/pad-utils.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
+
+const $rdf = require('rdflib')
+const ns = require('./ns')
+const PAD = $rdf.Namespace('http://www.w3.org/ns/pim/pad#')
+
+/* Get the chunbks of the noepad
```
Comment (michielbdejong) [inline src/pad-utils.js]: Typo

File: src/pad.js [hunk 7]
```diff
@@ -13,15 +15,22 @@ var UI = {
...
   rdf: $rdf,
   store: require('./store'),
   widgets: require('./widgets')
 }
-const kb = UI.store
-const ns = UI.ns
 
 const utils = require('./utils')
 
+// For back-compatiuble API only until next major release: @@ todo: remove these three
```
Comment (michielbdejong) [inline src/pad.js]: typo

File: src/participation.js (hunk 3 above)
Comment (SharonStrats) [reply src/participation.js]: If you want I can take a look at this, as I was working on pad.js testing?  And there were also some complexity warnings that I had either resolved or was working on.  Maybe I can combine these into one change.

Comment (michielbdejong) [reply src/participation.js]: If you're offering, I won't say no! :) https://github.com/solid/solid-ui/issues/210#event-3056026823

**Comment by michielbdejong** [comment]
@timbl can this be closed now that #211 is merged?

**Comment by michielbdejong** [comment]
See example usage here: http://solid.github.io/solid-ui/examples/notepad/

**Comment by megoth** [comment]
> @timbl can this be closed now that #211 is merged?

I'll await reviewing until this has a response.

**Comment by angelo-v** [comment]
If I understand this correctly, this PR has been superseded.
